src/types/fieldExtractor.ts

```diff
@@ -154,20 +166,32 @@ const idStrategy: StrategyExtractor = (contract) => {
   if (idExtractorOverride === IdExtractorTypes.USE_TITLE_EXTRACTOR) {
     return titleStrategy(contract);
   }
-  const extractor = idExtractors[idExtractorOverride];
+  let extractor = null;
+  if (typeof idExtractorOverride === 'string'){
+    extractor = idExtractors[idExtractorOverride];
+  }
+
   if (!extractor) {
     throw new Error('no other id extraction options yet')
   }
-  return extractor;
+  return extractor as Extractor;
 }
 
-export const resolveEthereumTrackId: Resolver = (nft, contract) => {
+export const resolveTrackId: Resolver = (nft, contract) => {
+  if (isParameterizedExtractor(contract.typeMetadata?.overrides.extractor?.id)){
+    const type = (contract.typeMetadata?.overrides.extractor?.id as ParameterizedExtractorType<IdExtractorTypes>)
+    const extractorType = type.extractor;
+    const extractor = idExtractors[extractorType]({ ...type.params, nftFactory: contract }) as Extractor;
+    return extractor(nft)
+  }
+
```

Comment:
think this code should go inside the idStrategy method?

(can then use the idExtractorOverride variable there and undefined/null check in there too instead of doing it twice)

also, some variable name change suggestions - maybe `ParameterizedExtractorConfig` is cleaner than `ParameterizedExtractorType`:

```suggestion
  if (isParameterizedExtractor(idExtractorOverride)){
    const config = (idExtractorOverride as ParameterizedExtractorConfig<IdExtractorTypes>)
    const extractorType = config.extractor;
    const extractor = idExtractors[extractorType]({ ...config.params, nftFactory: contract }) as Extractor;
    return extractor(nft)
  }

```

Reply:
ah right, cool. i guess the confusion is that this function resolveTrackId is basically the trackIdStrategy and idStrategy is actually more like trackIdSuffixStrategy - maybe should rename them as such?